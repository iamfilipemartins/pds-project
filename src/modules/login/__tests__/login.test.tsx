import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Login from '..';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router', () => ({
  ...(jest.requireActual('react-router') as any),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return {};
    }),
    useDispatch: jest.fn().mockImplementation(() => {
      return {};
    }),
  };
});

describe('<Login />', () => {
  const initialState = {
    country: {
      countrySelected: {
        mapData: {
          ABBREV: '',
          CONTINENT: '',
          FORMAL_EN: '',
          GDP_MD_EST: 0,
          GDP_YEAR: 0,
          ISO_A2: '',
          ISO_A3: '',
          NAME: '',
          NAME_LONG: '',
          POP_EST: 0,
          POP_RANK: 0,
          POP_YEAR: 0,
          REGION_UN: '',
          SUBREGION: '',
        },
        details: undefined,
      },
    },
    user: {
      login: {},
      signup: {},
      emailRecovery: '',
    },
  };
  const mockStore = configureStore();
  test('should display a blank login form, with remember me checked by default', async () => {
    const store = mockStore(initialState);
    const { findByTestId } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    const loginScreen = await findByTestId('login');

    expect(loginScreen).toMatchSnapshot();
  });

  test('should allow entering a email', async () => {
    const store = mockStore(initialState);
    const { findByTestId } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    const loginScreen = await findByTestId('login');

    const email = await findByTestId('email');

    fireEvent.change(email, { target: { value: 'test@test.com' } });

    expect(loginScreen).toBeDefined();
    expect(email).toBeDefined();
  });

  test('should allow entering a password', async () => {
    const store = mockStore(initialState);
    const { findByTestId } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    const password = await findByTestId('password');
    const loginScreen = await findByTestId('login');

    fireEvent.change(password, { target: { value: 'password' } });

    expect(loginScreen).toBeDefined();
    expect(password).toBeDefined();
  });
});
