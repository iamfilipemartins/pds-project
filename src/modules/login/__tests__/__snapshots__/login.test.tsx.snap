// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should display a blank login form, with remember me checked by default 1`] = `
<div
  class="sc-gsDKAQ iFzJiF"
  data-testid="login"
>
  <div
    class="sc-dkPtRN crvUag"
    width="1024"
  >
    <img
      class="sc-bdvvtL"
    />
    <h1
      class="sc-hKwDye gGnQLo"
      width="1024"
    >
      Geolog
    </h1>
  </div>
  <div
    class="sc-furwcr anivc"
    width="1024"
  >
    <h1
      class="sc-jrQzAO gfpuAK"
    >
      Login
    </h1>
    <div
      class="sc-kDTinF bInWyk"
    >
      <p
        class="sc-crHmcD dKnwJW"
      >
        Email *
      </p>
      <input
        class="sc-iqseJM dhtnqU"
        data-testid="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="sc-kDTinF bInWyk"
    >
      <p
        class="sc-crHmcD dKnwJW"
      >
        Senha *
      </p>
      <input
        class="sc-iqseJM dhtnqU"
        data-testid="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="sc-iCfMLu jSwPHu"
      width="1024"
    >
      <p
        class="sc-pVTFL gVDUJb"
      >
        Esqueceu a senha?
      </p>
      <div
        class="sc-egiyK"
      >
        <button
          class="sc-bqiRlB dpOhwe"
          color="#FFFFFF"
          disabled=""
        >
          Entrar
        </button>
      </div>
    </div>
  </div>
  <div
    class="sc-eCImPb iCHfyu"
    width="1024"
  >
    <p
      class="sc-jRQBWg dEaWjf"
    >
      Ainda n√£o possui uma conta?
    </p>
    <p
      class="sc-gKclnd kKqYGw"
    >
      Clique aqui
    </p>
  </div>
</div>
`;
